src/ngraph/op/gather.cpp

```diff
@@ -82,3 +95,119 @@ void op::Gather::validate_and_infer_types()
+    throw ngraph_error("Not yet implemented");
+}
+
+const string op::v1::Gather::type_name{"Gather"};
+
+op::v1::Gather::Gather(const Output<Node>& params,
+                       const Output<Node>& indices,
+                       const Output<Node>& axes)
+    : Op({params, indices, axes})
+{
+    constructor_validate_and_infer_types();
+}
+
+void op::v1::Gather::validate_and_infer_types()
+{
+    const auto& input_rank = get_input_partial_shape(PARAMS).rank();
+    const auto& axis_shape = get_input_partial_shape(AXIS);
+    const auto& axis_rank = axis_shape.rank();
+
+    if (axis_rank.is_static())
+    {
+        NODE_VALIDATION_CHECK(this,
+                              static_cast<size_t>(axis_rank) == 1 &&
+                                  static_cast<size_t>(axis_shape[0]) == 1,
+                              "Axes input must have 1 element (shape: ",
+                              axis_shape,
+                              ").");
+    }
+
+    auto axis = get_axis();
+    if (input_rank.is_static() && axis != AXIS_NOT_SET_VALUE)
+    {
+        NODE_VALIDATION_CHECK(this,
+                              axis >= 0 && axis < static_cast<size_t>(input_rank),
+                              "The axis must => 0 and <= input_rank (axis: ",
+                              axis,
+                              ").");
+    }
+
+    element::Type result_et = get_input_element_type(PARAMS);
+    element::Type indices_et = get_input_element_type(INDICES);
+
+    const PartialShape& params_shape = get_input_partial_shape(PARAMS);
+    const PartialShape& indices_shape = get_input_partial_shape(INDICES);
+
+    PartialShape result_shape;
+    if (params_shape.rank().is_static() && indices_shape.rank().is_static() &&
+        axis != AXIS_NOT_SET_VALUE)
+    {
+        std::vector<Dimension> result_dims(static_cast<size_t>(params_shape.rank()) +
+                                           static_cast<size_t>(indices_shape.rank()) - 1);
+        size_t i = 0;
+        for (; i < static_cast<size_t>(axis); i++)
+        {
+            result_dims[i] = params_shape[i];
+        }
+        for (size_t j = 0; j < static_cast<size_t>(indices_shape.rank()); i++, j++)
+        {
+            result_dims[i] = indices_shape[j];
+        }
+        for (size_t j = static_cast<size_t>(axis) + 1; j < static_cast<size_t>(params_shape.rank());
+             i++, j++)
+        {
+            result_dims[i] = params_shape[j];
+        }
+
+        result_shape = PartialShape(result_dims);
+    }
+    else
+    {
+        result_shape = PartialShape::dynamic();
+    }
+
+    set_output_type(0, result_et, result_shape);
+}
+
+int64_t op::v1::Gather::get_axis() const
+{
+    int64_t axis = AXIS_NOT_SET_VALUE;
+    auto axes_input_node = input_value(AXIS).get_node_shared_ptr();
+    if (auto const_op = dynamic_pointer_cast<op::Constant>(axes_input_node))
+    {
+        axis = const_op->get_vector<int64_t>()[0];
+    }
+    else
+    {
+        axis = AXIS_NOT_SET_VALUE;
```

Comment:
I think this `else` branch is redundant since you've already initialized `axis` to `AXIS_NOT_SET_VALUE`.